### DISCUSSION TITLE
Inference time is more in release 6 vs release 4 for custom trained model yolov5m.pt

### DISCUSSION THREAD

**Discussion by myuvarajmp** [ultralytics/yolov5#5282] [Q&A]
Hi All,

I have custom trained model for both yolov5 release 4 and release 6 using yolov5m.pt weights given to them. When I check the inference using Nvidia T4 GPU I get inference for 4 images in batch mode at 0.062 in release 6 and 0.036 in release 4. What all things i need to check in detect.py and what could be possible issue.

**Comment by glenn-jocher** [comment]
@myuvarajmp I tested both in Colab on a P100 GPU and see similar speeds as expected. In practice your speeds may vary by hardware, software and drivers.

<img width="1231" alt="Screen Shot 2021-10-21 at 5 04 55 PM" src="https://user-images.githubusercontent.com/26833433/138305924-cf566d3c-da15-4238-b2ce-0080d2ae6a48.png">

**Reply by myuvarajmp** [reply]
--half did the trick. It works. Could you please explain what does --half does that it reduces the inference time.

**Reply by glenn-jocher** [reply]
@myuvarajmp oh yes. --half runs half-precision (FP16) inference on GPU. Default inference is at FP32.

[PyTorch Hub](https://github.com/ultralytics/yolov5/issues/36) models, and train.py, use mixed precision context managers that automatically run at FP16 if possible. detect.py and val.py default to FP32 and have to opt-in to FP16 with --half.
